cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/visitor/ReleaseVisitor.java

```diff
@@ -177,66 +213,67 @@ private static void updateMetadata(
-                                rootEffectivePeriod,
-                                latestFromTxServer,
-                                experimentalBehavior,
-                                current,
-                                releasedResources));
-                    } else {
-                        // if missing throw because it's an owned resource
-                        throw new ResourceNotFoundException(String.format(
-                                "Resource with URL '%s' is Owned by this repository and referenced by resource '%s', but no active version was found on the server.",
-                                preReleaseReference, artifactAdapter.getUrl()));
+                        updatedComponents.addAll(updateAllComponents(
+                                latest.get(), latestFromTxServer, updatedResourceReferences, endpoint));
                     }
-                } else if (!alreadyUpdated.isPresent()) {
-                    // if it's a not-owned component just try to get the latest active version
-                    VisitorHelper.tryGetLatestVersionWithStatus(preReleaseReference, repository, ACTIVE)
-                            .ifPresent(latestActive ->
-                                    // check if it's experimental
-                                    checkNonExperimental(latestActive.get(), experimentalBehavior, repository));
                 }
             }
         }
-        return resourcesToUpdate;
+        return updatedComponents;
     }
 
+    private Optional<IKnowledgeArtifactAdapter> latestComponentRespectingVersions(
+            String preReleaseReference, IEndpointAdapter endpoint, boolean latestFromTxServer, boolean isOwned) {
+        Optional<IKnowledgeArtifactAdapter> latest = Optional.empty();
+        var resourceType = Canonicals.getResourceType(preReleaseReference);
+        var prereleaseReferenceVersion = Canonicals.getVersion(preReleaseReference);
+        if (isOwned) {
+            // get the latest version regardless of status because we own the resource so Drafts are ok (they'll be
+            // updated as part of $release)
+            latest = VisitorHelper.tryGetLatestVersion(preReleaseReference, repository);
+        } else if (resourceType != null
+                && resourceType.equals(VALUESET)
+                && prereleaseReferenceVersion == null
+                && latestFromTxServer) {
+            // ValueSets we don't own go to the Tx Server if latestFromTxServer is true
+            latest = terminologyServerClient
+                    .getResource(endpoint, preReleaseReference, this.fhirVersion())
+                    .map(r -> (IKnowledgeArtifactAdapter) createAdapterForResource(r));
+        } else {
+            // get the latest ACTIVE version only because it's NOT owned
+            latest = VisitorHelper.tryGetLatestVersionWithStatus(preReleaseReference, repository, ACTIVE);
+        }
+        return latest;
+    }
```

Comment:
@sliver007 Wanted to check that the logic here is correct, for components, if `latestFromTxServer` is true it should only query the Tx Server if:

1. The component is NOT owned
2. The reference does not have a version
3. The component is referencing a ValueSet

Reply:
Is there some level of 'authoritativeSource' awareness that is needed here - similar to other areas of $release?

Reply:
yes there is, thanks 😬

Reply:
we can't get the actual authoritative source though, we can only check to see if the URL on the endpoint matches the URL in the reference and then 🤞🏾

Reply:
As per our Slack discussion left it as is and added a comment explaining the assumption 👍🏾 (we assume the caller knows what they're doing, at worst the VSets are just not resolved)